### PR TITLE
Switch to File Based Dynamic Config Client

### PR DESCRIPTION
**Pull request by yycptt**

### CODE REVIEW

File: common/service/config/config.go [hunk 1]
```diff
@@ -57,6 +57,9 @@ type (
 		ElasticSearch elasticsearch.Config `yaml:"elasticsearch"`
 		// PublicClient is config for connecting to cadence frontend
 		PublicClient PublicClient `yaml:"publicClient"`
+		// DCClientConfig is the config for setting up the file based dynamic config client
+		// Filepath should be relative to the root directory
+		DCClientConfig dynamicconfig.FileBasedClientConfig `yaml:"dcClientConfig"`
```
Comment (vancexu) [inline common/service/config/config.go]: Let's rename to "DynamicConfigClient", 
dc may not sounds familiar to customers, and team may confuse it with data center.

File: config/base.yaml [hunk 2]
```diff
@@ -1,3 +1,7 @@
 log:
   stdout: true
   level: info
+
+dcClientConfig:
```
Comment (vancexu) [inline config/base.yaml]: Let's move it to `development.yaml`. 
Since this is good to have but still optional

File: cmd/server/server.go [hunk 3]
```diff
@@ -111,7 +112,11 @@ func (s *server) startService() common.Daemon {
 		log.Fatalf("error creating ringpop factory: %v", err)
 	}
 
-	params.DynamicConfig = dynamicconfig.NewNopClient()
+	params.DynamicConfig, err = dynamicconfig.NewFileBasedClient(&s.cfg.DCClientConfig, params.Logger.WithTags(tag.Service(params.Name)), s.doneC)
+	if err != nil {
+		log.Printf("error creating file based dynamic config client: %v", err)
```
Comment (vancexu) [inline cmd/server/server.go]: I would make it:
"cannot create filed based dynamic config client, so will use no-op config client instead. error: %v"

File: config/dynamicconfig.yaml [hunk 4]
```diff
@@ -0,0 +1,37 @@
+# Use this file to override the default dynamic config value (they are specified
```
Comment (vancexu) [inline config/dynamicconfig.yaml]: Let's move sample config to path `config/dynamic_config/dev.yaml`

File: common/service/config/config.go (hunk 1 above)
Comment (yycptt) [reply common/service/config/config.go]: Update.

File: config/base.yaml (hunk 2 above)
Comment (yycptt) [reply config/base.yaml]: Done

File: cmd/server/server.go (hunk 3 above)
Comment (yycptt) [reply cmd/server/server.go]: Done.

File: config/dynamicconfig.yaml (hunk 4 above)
Comment (yycptt) [reply config/dynamicconfig.yaml]: Done

**Pull request merged**